connect/api_test.go

```diff
@@ -82,3 +82,44 @@ func TestUpToDateOkay(t *testing.T) {
 		t.Error("Expecting UpToDate()==true, got false")
 	}
 }
+
+func TestGetProduct(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		w.Write(readTestFile("product.json", t))
```

Comment:
if readTestFile() calls Fatal(), it won't actually stop the test, because it's being called from the handler goroutine and not the test goroutine. See https://golang.org/pkg/testing/#T.FailNow

Reply:
The test will fail with:
```
=== RUN   TestGetProduct
    helpers_test.go:14: 
        api_test.go:92: open ../testdata/product.json: no such file or directory
    api_test.go:100: Get "http://127.0.0.1:38759/connect/systems/products?arch=x86_64&identifier=SLES&version=15.2": EOF
```
I think it's good enough.

Reply:
okay with me.